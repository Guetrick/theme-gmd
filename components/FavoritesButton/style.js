/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { css } from 'glamor';
import colors from 'Styles/colors';

const button = css({
  display: 'block',
  position: 'relative',
  background: '#fff',
  borderRadius: '50%',
  padding: 0,
  fontSize: 20,
  lineHeight: 1,
  color: colors.accent,
  outline: 0,
  zIndex: 15, // Should overlap the gallery bullets
}).toString();

const ripple = css({
  padding: 6,
}).toString();

export default {
  button,
  ripple,
};
