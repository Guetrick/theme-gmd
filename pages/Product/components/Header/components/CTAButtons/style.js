/**
 * Copyright (c) 2017 - present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { css } from 'glamor';
import variables from 'Styles/variables';

const buttons = css({
  position: 'absolute',
  right: variables.gap.big,
  top: -30,
  display: 'flex',
  justifyContent: 'flex-end',
  alignItems: 'center',
}).toString();

const favButton = css({
  marginRight: variables.gap.big,
  boxShadow: '0 8px 13px rgba(0, 0, 0, 0.25)',
  zIndex: 15, // Prevents the icons to be visible outside of the circle
  fontSize: 24,
}).toString();

const ripple = css({
  padding: 8,
}).toString();

export default {
  buttons,
  favButton,
  ripple,
};
