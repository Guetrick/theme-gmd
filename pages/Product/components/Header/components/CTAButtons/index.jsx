/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import FavoritesButton from 'Components/FavoritesButton';
import AddToCartButton from './components/AddToCartButton';
import styles from './style';

/**
 * Renders CTA buttons for product page (add to cart + toggle favorites).
 *
 * @returns {XML}
 * @constructor
 */
const CTAButtons = () => (
  <div className={styles.buttons}>
    <FavoritesButton
      className={styles.favButton}
      rippleClassName={styles.ripple}
    />
    <AddToCartButton />
  </div>
);

export default CTAButtons;
