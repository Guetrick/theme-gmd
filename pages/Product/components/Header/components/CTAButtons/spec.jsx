/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import { shallow } from 'enzyme';
import FavoritesButton from 'Components/FavoritesButton';
import CTAButtons from './index';
import AddToCartButton from './components/AddToCartButton';

describe('CTAs (product header)', () => {
  describe('Rendering', () => {
    it('should render rating when data is available', () => {
      const component = shallow(<CTAButtons />);
      expect(component).toMatchSnapshot();
      expect(component.find(FavoritesButton).exists()).toBe(true);
      expect(component.find(AddToCartButton).exists()).toBe(true);
    });
  });
});

