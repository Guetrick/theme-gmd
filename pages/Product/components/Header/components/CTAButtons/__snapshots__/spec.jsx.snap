// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CTAs (product header) Rendering should render rating when data is available 1`] = `
<div
  className="css-7azkrx"
>
  <Connect(FavoritesButton)
    className="css-16tssa1"
    rippleClassName="css-1tdptiz"
  />
  <Connect(AddToCartButton) />
</div>
`;
